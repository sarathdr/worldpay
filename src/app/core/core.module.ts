import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
