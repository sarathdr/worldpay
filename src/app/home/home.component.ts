import { Component, Input, OnInit } from '@angular/core';
import { Cart } from '../core/models/cart.interface';

@Component({
  selector: 'home',
  template: `
    <div class="cart">

      <table mat-table [dataSource]="cart.products" class="mat-elevation-z8">

        <ng-container matColumnDef="name">
          <th mat-header-cell *matHeaderCellDef> Products</th>
          <td mat-cell *matCellDef="let element"> {{ element.name }} </td>
          <td mat-footer-cell *matFooterCellDef> Total</td>
        </ng-container>

        <ng-container matColumnDef="price">
          <th mat-header-cell *matHeaderCellDef> Price</th>
          <td mat-cell *matCellDef="let element"> {{ element.price | currency: cart.currency }} </td>
          <td mat-footer-cell *matFooterCellDef> <b>{{ cart.total | currency: cart.currency }}</b></td>

        </ng-container>

        <tr mat-header-row *matHeaderRowDef="['name', 'price']"></tr>
        <tr mat-row *matRowDef="let row; columns: ['name', 'price'];"></tr>
        <tr mat-footer-row *matFooterRowDef="['name', 'price']; sticky: true"></tr>
      </table>
      <button mat-raised-button color="primary" routerLink="/payment">Continue to payment</button>
    </div>
  `,
  styles: [
      `
    .cart {
        margin: 0 auto;
        display: flex;
        flex-direction: column;
        width: 400px;
    }
    .cart table {
        width: 400px;
        margin-bottom: 30px;
    }
    `
  ]
})
export class HomeComponent {
  @Input() cart: Cart;
}
