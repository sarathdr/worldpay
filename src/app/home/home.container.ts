import { Component } from '@angular/core';
import { Observable } from 'rxjs';
import { CartService } from '../core/services/cart.service';
import { Cart } from '../core/models/cart.interface';

@Component({
  selector: 'home-page',
  template: `
    <home
        [cart]="cart$ | async">
    </home>
  `,
})
export class HomeContainer {
  cart$: Observable<Cart>;

  constructor(cartService: CartService) {
    this.cart$ = cartService.getCart();
  }

}
