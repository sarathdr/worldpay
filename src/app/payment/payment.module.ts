import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthorizeComponent } from './authorize/authorize.component';
import { AuthorizeContainer } from './authorize/authorize.container';
import { ConfirmComponent } from './confirm/confirm.component';
import { ConfirmContainer } from './confirm/confirm.container';
import { RefundComponent } from './refund/refund.component';
import { RefundContainer } from './refund/refund.container';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { MatButtonModule, MatFormFieldModule, MatInputModule, MatSelectModule } from '@angular/material';
import { AuthorizeFormComponent } from './authorize/authorize-form.component';
import { ConfirmPaymentGuard } from './guards/confirm-payment.guard';
import { ExpiryDateModule } from '../core/directives/expiry-date.module';
import { RefundPaymentGuard } from './guards/refund-payment.guard';
import { RefundSuccessComponent } from './refund/refund-success.component';

@NgModule({
  declarations: [
    AuthorizeComponent,
    AuthorizeFormComponent,
    AuthorizeContainer,
    ConfirmComponent,
    ConfirmContainer,
    RefundComponent,
    RefundSuccessComponent,
    RefundContainer
  ],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule.forChild([
      {
        path: '',
        component: AuthorizeContainer
      },
      {
        path: 'confirm',
        component: ConfirmContainer,
        canActivate:[ConfirmPaymentGuard],
        canDeactivate:[ConfirmPaymentGuard]
      },
      {
        path: 'refund',
        component: RefundContainer,
        canActivate:[RefundPaymentGuard],
        canDeactivate:[RefundPaymentGuard]
      }
    ]),
    MatFormFieldModule,
    MatInputModule,
    MatSelectModule,
    ReactiveFormsModule,
    MatButtonModule,
    ExpiryDateModule
  ],
  providers:[
    ConfirmPaymentGuard,
    RefundPaymentGuard
  ]
})
export class PaymentModule { }
