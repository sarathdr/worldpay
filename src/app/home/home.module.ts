import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HomeComponent } from './home.component';
import { RouterModule } from '@angular/router';
import { HomeContainer } from './home.container';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';


@NgModule({
  declarations: [HomeContainer, HomeComponent],
  imports: [
    CommonModule,
    RouterModule.forChild([
      {
        path: '',
        component: HomeContainer
      }
    ]),
    MatListModule,
    MatButtonModule,
    MatTableModule
  ]
})
export class HomeModule {
}
