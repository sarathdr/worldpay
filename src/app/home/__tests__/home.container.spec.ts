import { TestBed } from '@angular/core/testing';
import { HomeComponent } from '../home.component';
import { Currency } from '../../core/models/payment.interface';
import { Component, Input } from '@angular/core';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { HomeContainer } from '../home.container';
import { CartService } from '../../core/services/cart.service';

describe('HomeComponent', () => {

  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MatListModule,
        MatButtonModule,
        MatTableModule
      ],
      declarations: [
        HomeContainer,
        MockHomeComponent
      ],
      providers: [
        CartService
      ]
    });

    fixture = TestBed.createComponent(HomeContainer);
  });

  it('should render the component correctly', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  @Component({
    selector: 'home',
    template: `
    <div>Home: {{ cart | json }}</div>
    `
  })
  class MockHomeComponent {
    @Input() cart;
  }

});
