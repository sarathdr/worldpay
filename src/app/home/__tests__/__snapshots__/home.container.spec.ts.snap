// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render the component correctly 1`] = `
<home-page
  cart$={[Function Observable]}
>
  <home
    ng-reflect-cart="[object Object]"
  >
    <div>
      Home: {
    "products": [
      {
        "name": "Pen",
        "price": 10
      },
      {
        "name": "Pencil",
        "price": 12
      }
    ],
    "total": 22,
    "currency": "GBP"
  }
    </div>
  </home>
</home-page>
`;
